tests/unit/backend/data/test_GroceryService.py

```diff
@@ -613,6 +615,37 @@ def test_rebuildCache(self, mockNeutronWorkspaceName):
         assert self.instance._loadedGroupings == {}
         assert self.instance._loadedInstruments == {}
 
+    def test_workspaceTagFunctions(self):
+        wsname = "testWorkspaceTags"
+        tagValue = "none"
+        properties = list(WorkspaceMetadata.schema()["properties"].keys())
+        logName = properties[0]
+
+        # Make sure correct error is thrown if workspace does not exist
+        with pytest.raises(RuntimeError) as e1:
+            tagResult = self.instance.getWorkspaceTag(workspaceName=wsname, logname=logName)
+        assert f"Workspace {wsname} does not exist" in str(e1.value)
+
+        with pytest.raises(RuntimeError) as e2:
+            self.instance.setWorkspaceTag(workspaceName=wsname, logname=logName, logvalue=tagValue)
+        assert f"Workspace {wsname} does not exist" in str(e2.value)
+
+        # Make sure default tag value is unset
+        self.create_dumb_workspace(wsname)
+        tagResult = self.instance.getWorkspaceTag(workspaceName=wsname, logname=logName)
+        assert tagResult == "unset"
+
+        # Verify the set tag value is correct
+        self.instance.setWorkspaceTag(workspaceName=wsname, logname=logName, logvalue=tagValue)
+        tagResult = self.instance.getWorkspaceTag(workspaceName=wsname, logname=logName)
+        assert tagResult == tagValue
+
+        # Make sure that you can update the tag value
+        tagValue = "exists"
```

Comment:
Up above, import
``` python
from snapred.backend.dao.WorkspaceMetadata import WorkspaceMetadata, diffcal_metadata_state_list
```
then here use a loop over all of the values, setting each one.